test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala

```diff
@@ -0,0 +1,211 @@
+package scala
+package tools.nsc
+package reporters
+
+import java.io.{ByteArrayOutputStream, StringReader, BufferedReader, PrintStream, PrintWriter}
+import org.junit.Assert._
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+import scala.reflect.internal.util._
+
+
+@RunWith(classOf[JUnit4])
+class ConsoleReporterTest {
+  val noPos = NoPosition
+  val source = "Test_ConsoleReporter"
+  val batchFile = new BatchSourceFile(source, "For testing".toList)
+  val posWithSource = new OffsetPosition(batchFile, 4)
+  val content = posWithSource.lineContent
+
+  def createConsoleReporter(inputForReader: String, writerOut: ByteArrayOutputStream, echoWriterOut: ByteArrayOutputStream = null): ConsoleReporter = {
+      val writer = new PrintWriter(writerOut)
+      val reader = new BufferedReader(new StringReader(inputForReader))
+      
+      /** create reporter with the same writer and echoWriter */
+      if(echoWriterOut == null)
+        new ConsoleReporter(new Settings(), reader, writer) 
+      else
+      {
+        val echoWriter = new PrintWriter(echoWriterOut)
+        new ConsoleReporter(new Settings(), reader, writer, echoWriter)
+      }
+  }
+
+  /** Create a reporter */
+  val writerOut = new ByteArrayOutputStream()
+  val echoWriterOut = new ByteArrayOutputStream()
+  val reporter = createConsoleReporter("s", writerOut, echoWriterOut)
+
+  /** Change maxerrs and maxwarns */
+  reporter.settings.maxerrs.value = 50
+  reporter.settings.maxwarns.value = 25
+
+
+  def testHelper(checkBlank: Boolean, hasPos: Boolean, message: String, severity: String): Unit = {
+    if(checkBlank) assertEquals("", writerOut.toString)
+    else {
+      if(!hasPos) assertEquals(severity + message, writerOut.toString.lines.next)
+      else {
+        val it = writerOut.toString.lines
+        assertEquals(source + ":1: " + severity + message, it.next)
+        assertEquals(content, it.next)
+        assertEquals("    ^", it.next)
+      }
+    }
+    writerOut.reset
+  }
```

Comment:
I think I was trying to avoid repeating the pos info. My second example reads well; you'd only need arrow syntax to ignore the pos in the test function.

But yes, you could just:

`testHelper(myPos, "test")(reporter.print(myPos, "test", INFO))`

with by-name param instead of a function:

`def testHelper(...)(test: => Unit)`

Either way, I think it's worth the trouble, because they want to do more tricks with reporters, like filtering, so it's likely your test will get exercised and expanded.